http/oidc/src/main/java/org/wildfly/security/http/oidc/OidcHttpFacade.java

```diff
@@ -205,9 +205,10 @@ public String getMethod() {
             @Override
             public String getURI() {
                 try {
-                    return URLDecoder.decode(request.getRequestURI().toString(), "UTF-8");
-                } catch (UnsupportedEncodingException e) {
-                    throw log.failedToDecodeRequestUri(e);
+                    URI decode = new URI(request.getRequestURI().toString());
```

Comment:
Hello @kanha-gupta @fjuma , I am not sure if we can remove the decoding of the URL this way. While this solves the problem of spaces in URL params, some other characters might not get decoded properly with using URI.toString method. For now, I have asked the reporter for more specific reproducer

Reply:
Okay, thanks :)